atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/DefaultShardProgressUpdaterTest.java

```diff
@@ -0,0 +1,269 @@
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class DefaultShardProgressUpdaterTest {
+    @Mock
+    private BucketProgressStore bucketProgressStore;
+
+    @Mock
+    private SweepQueueProgressUpdater sweepQueueProgressUpdater;
+
+    @Mock
+    private SweepBucketRecordsTable recordsTable;
+
+    @Mock
+    private SweepBucketPointerTable sweepBucketPointerTable;
+
+    private DefaultShardProgressUpdater shardProgressUpdater;
+
+    @BeforeEach
+    public void setUp() {
+        shardProgressUpdater = new DefaultShardProgressUpdater(
+                bucketProgressStore, sweepQueueProgressUpdater, recordsTable, sweepBucketPointerTable);
+    }
+
+    @ParameterizedTest
+    @MethodSource("buckets")
+    public void wrapsAndRethrowsExceptionOnAbsenceOfTimestampRangeRecords(Bucket bucket) {
+        when(sweepBucketPointerTable.getStartingBucketsForShards(ImmutableSet.of(bucket.shardAndStrategy())))
+                .thenReturn(ImmutableSet.of(bucket));
+        NoSuchElementException underlyingException = new NoSuchElementException();
+        when(recordsTable.getTimestampRangeRecord(bucket.bucketIdentifier())).thenThrow(underlyingException);
+
+        assertThatLoggableExceptionThrownBy(() -> shardProgressUpdater.updateProgress(bucket.shardAndStrategy()))
+                .isInstanceOf(SafeIllegalStateException.class)
+                .hasLogMessage("Timestamp range record not found. If this has happened for bucket 0, this is possible"
+                        + " when autoscaling sweep is initializing itself. Otherwise, this is potentially indicative of"
+                        + " a bug in auto-scaling sweep. In either case, we will retry.")
+                .hasExactlyArgs(SafeArg.of("queriedBucket", bucket.bucketIdentifier()))
+                .hasCause(underlyingException);
+
+        verify(sweepBucketPointerTable, never()).updateStartingBucketForShardAndStrategy(bucket);
+        verify(sweepQueueProgressUpdater, never()).progressTo(eq(bucket.shardAndStrategy()), anyLong());
+        verify(bucketProgressStore, never()).deleteBucketProgress(any());
+    }
+
+    @ParameterizedTest
+    @MethodSource("buckets")
+    public void doesNotUpdateProgressOnUnstartedBucket(Bucket bucket) {
+        when(sweepBucketPointerTable.getStartingBucketsForShards(ImmutableSet.of(bucket.shardAndStrategy())))
+                .thenReturn(ImmutableSet.of(bucket));
+        when(bucketProgressStore.getBucketProgress(bucket)).thenReturn(Optional.empty());
+        when(recordsTable.getTimestampRangeRecord(bucket.bucketIdentifier()))
+                .thenReturn(TimestampRange.of(0L, SweepQueueUtils.minTsForCoarsePartition(8)));
```

Comment:
Consider: having a different timestamp than 0L to show that the progressTo is start - 1, rather than just -1L (Which would be a bug that's possible for someone to actually do)

Reply:
Yep, will change to use `SweepQueueUtils.minTsForCoarsePartition(3)`